rclcpp/include/rclcpp/strategies/static_memory_strategy.hpp

```diff
@@ -28,6 +28,12 @@ namespace memory_strategies
 namespace static_memory_strategy
 {
 
+template<size_t max_subscriptions_ = 10,
+size_t max_services_ = 10,
+size_t max_clients_ = 10,
+size_t max_executables_ = 1,
+size_t max_guard_conditions_ = 2,
+size_t pool_size_ = 1024>
```

Comment:
These arguments specify the sizes of arrays that are class members. Therefore they need to be const expressions.

I could make these constructor arguments. Then I would have to malloc the object pool arrays in the constructor.

It is slightly preferable to know the pool sizes at compile time but either one is acceptable for real-time performance.